### ISSUE TITLE
OOM event of exec is the same as OOM event of container main process

### ISSUE THREAD

**Issue by garagatyi** [docker/docker#18433]
It is not possible to tell the difference between OOM events of exec and container main process. Example of these events:
OOM of exec
```
{"status":"oom","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240479,"timeNano":1449240479251111202}
```
OOM  of main process of container
```
{"status":"oom","id":"52dfe5cc7cf6d83a2fe5ce9e474cb5023f35146dd6097808771bb94a609d3a50","from":"800c5e14c91a","time":1449240569,"timeNano":1449240569798358395}
```
Is it a normal behavior or these events should differ?

To reproduce:
```bash
rm -rf /tmp/oomeventstest && \
mkdir /tmp/oomeventstest && \
cd /tmp/oomeventstest && \
wget https://gist.githubusercontent.com/garagatyi/ec0c1a735a8e6b355b38/raw/de3154ded3a2787fae3a93837b5e861919bde41c/Dockerfile && \
docker build .
```
Then in other terminal:
```bash
curl -s http://localhost:2375/events
```
Then in first terminal:
```bash
docker run -d -m 128M {imageId}
docker exec -d {containerId} /script.sh
```
Wait a minute or two, then:
```bash
docker run -d -m 128M {imageId} /script.sh
```
Wait a minute or two.

Output of curl:
```
{"status":"create","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240345,"timeNano":1449240345443705479}
{"status":"start","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240345,"timeNano":1449240345798029258}
{"status":"exec_create: /script.sh ","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240448,"timeNano":1449240448515346498}
{"status":"exec_start: /script.sh ","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240448,"timeNano":1449240448515878864}
{"status":"oom","id":"32c6100a9e288e6c68ff9b0f072dab2f9ee5568dbbf1e50df719e2e0bcb8d776","from":"800c5e14c91a","time":1449240479,"timeNano":1449240479251111202}
{"status":"create","id":"52dfe5cc7cf6d83a2fe5ce9e474cb5023f35146dd6097808771bb94a609d3a50","from":"800c5e14c91a","time":1449240539,"timeNano":1449240539448591588}
{"status":"start","id":"52dfe5cc7cf6d83a2fe5ce9e474cb5023f35146dd6097808771bb94a609d3a50","from":"800c5e14c91a","time":1449240539,"timeNano":1449240539823100403}
{"status":"oom","id":"52dfe5cc7cf6d83a2fe5ce9e474cb5023f35146dd6097808771bb94a609d3a50","from":"800c5e14c91a","time":1449240569,"timeNano":1449240569798358395}
{"status":"die","id":"52dfe5cc7cf6d83a2fe5ce9e474cb5023f35146dd6097808771bb94a609d3a50","from":"800c5e14c91a","time":1449240569,"timeNano":1449240569897126585}
```

**Comment by thaJeztah** [comment]
There's an open proposal to add more information / meta-data to events; https://github.com/docker/docker/issues/18164. (If possible) we could add the PID of the process that OOM'd, would that work for your use case?

**Comment by garagatyi** [comment]
I love idea to add more metadata to events. And yes adding PID will help in current use case.

**Comment by thaJeztah** [comment]
ping @calavera wdyt? would that be possible, and a reasonable thing to add to the event as meta-data?

**Comment by garagatyi** [comment]
Another option is to add exec ID to this event.

**Comment by LK4D4** [comment]
I think this was fixed in #18888
Feel free to open new issue if it's not.
